### ISSUE TITLE
goto definition in external rust source hangs helix

### ISSUE THREAD

**Issue by CptPotato** [helix-editor/helix#2187]
I seem to be unable to jump to a definition if it's not in the current workspace. When doing so, helix locks up for me in an infinite loop, increasing memory over time.

I'm not sure what exactly causes this, I recently updated helix, rust and rust-analyzer, so it might have something to do with that.
I'll check if I can reproduce it on a different machine later.

---

### Minimal repro

Jumping to `Some`:

```rs
fn main {
    let _ = Some(1);
}
```

os: Win10 x64
hx: `helix 22.05-dev (1525e3c6)`

**Comment by CptPotato** [comment]
After a quick bisect, it seems like this was introduced with https://github.com/helix-editor/helix/commit/c8cfd0b1a0da071618a9efc71ac5735d6147a0ca

**Comment by archseer** [comment]
I'll roll back c8cfd0b1a0da071618a9efc71ac5735d6147a0ca for now

**Comment by CptPotato** [comment]
> I'll roll back [c8cfd0b](https://github.com/helix-editor/helix/commit/c8cfd0b1a0da071618a9efc71ac5735d6147a0ca) for now

Makes sense, considering the other issue related to this commit/PR.

If I can do anything to debug or provide more info, let me know.

**Comment by the-mikedavis** [comment]
Whoops! 😅

Just as well with the revert, I just found some buggy behavior with the merge anyways. Do you have a custom `languages.toml` @CptPotato? I don't have this issue on that commit but I think it's because I don't have a custom `languages.toml` to merge.

**Comment by CptPotato** [comment]
Yes I do have a custom `languages.toml`. I think the only thing I changed was setting all `auto-format = true` to `false`.

Here's the language config, if that helps: [languages.toml](https://github.com/helix-editor/helix/files/8521599/languages.toml.txt)

**Comment by the-mikedavis** [comment]
Ok I'll take a look at that in #2188 

Let's close this one out now since c8cfd0b is reverted (https://github.com/helix-editor/helix/commit/35d2693630a4ec29a654704bc4be47badb8d6070)
